### ISSUE TITLE
How many steps for RealGenv2?

### ISSUE THREAD

**Issue by mihu0410** [yejy53/RealGen#11]
1. Under the default configuration config/realgen.py:zimage_base_fast_guard, roughly how many training steps are recommended for the open-source LoRA? 
2. Furthermore, for ZImage-Turbo using the default config zimage_turbo_fast_guard, approximately how many steps should it run? I'm asking because I observed some hacking issues during experiments, and I suspect it might be caused by excessive training.
3. Thanks for this excellent works！

**Comment by garychan22** [comment]
Same question here. And, can the authors release the reward curve versue the increasing global step? 
Thanks!

**Comment by yunncheng** [comment]
We recommend limiting the training to within 200 steps; in our implementation, we specifically opt for 160 steps. We observe that reward hacking starts to manifest beyond 300 steps. The reward curve versue the increasing global step is shown below:

<img width="1775" height="741" alt="Image" src="https://github.com/user-attachments/assets/22a52345-eca4-42fa-b2c6-a1e06f18d4c4" />

**Comment by mihu0410** [comment]
Get it, Thanks！

**Comment by garychan22** [comment]
> We recommend limiting the training to within 200 steps; in our implementation, we specifically opt for 160 steps. We observe that reward hacking starts to manifest beyond 300 steps. The reward curve versue the increasing global step is shown below:
> 
> <img alt="Image" width="1775" height="741" src="https://private-user-images.githubusercontent.com/116631334/579349627-22a52345-eca4-42fa-b2c6-a1e06f18d4c4.png?jwt=eyJ0eXAiOiJKV1QiLCJhbGciOiJIUzI1NiJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NzY0MTUxOTIsIm5iZiI6MTc3NjQxNDg5MiwicGF0aCI6Ii8xMTY2MzEzMzQvNTc5MzQ5NjI3LTIyYTUyMzQ1LWVjYTQtNDJmYS1iMmM2LWExZTA2ZjE4ZDRjNC5wbmc_WC1BbXotQWxnb3JpdGhtPUFXUzQtSE1BQy1TSEEyNTYmWC1BbXotQ3JlZGVudGlhbD1BS0lBVkNPRFlMU0E1M1BRSzRaQSUyRjIwMjYwNDE3JTJGdXMtZWFzdC0xJTJGczMlMkZhd3M0X3JlcXVlc3QmWC1BbXotRGF0ZT0yMDI2MDQxN1QwODM0NTJaJlgtQW16LUV4cGlyZXM9MzAwJlgtQW16LVNpZ25hdHVyZT1jMjlmMDRlY2Q5OWI4ZDRjMWYyYzBkMDI5MjVlNzQxZmZlYTUwNGM5YmI5YjliZWIwNGQ1ODFiZWU1ZmE3ZmViJlgtQW16LVNpZ25lZEhlYWRlcnM9aG9zdCZyZXNwb25zZS1jb250ZW50LXR5cGU9aW1hZ2UlMkZwbmcifQ.2iGbalf3cf2wqlda2wG4xrGMOC_y7OO28p9q-adIPXc">

Thanks for the kindly reply!

I have noticed that in RealGenV2 the in-training evaluation using the updated reward models to measure the generated images, should we use the frozen ones to achieve this goal since we cannot sure whether the updataed reward models is trained in the right direction due to its training instability or overfitting to the training dataset, or there is something I misunderstood?
